_posts/2019-11-27-update-sitemaps-during-heroku-deploy.markdown

```diff
@@ -0,0 +1,122 @@
+  add '/contact_us', :changefreq => 'weekly'
+end
+```
+
+As you can see, inside the `Sitemap.create` block we need to add all routes of our site.
+
+If your site has only static content, your job here it's done. You just need to use the rails task from the sitemap_generator itself `rake sitemap:refresh` and a `sitemap.xml` file will be added to the `public` folder.
+
+But in my case new routes will be added every time that someone writes a new article. And as we know, the blog it's a Jekyll application,
+so we can't get the routes using the rake task provided by the `sitemap_generator` gem.
+
+## Generating the sitemap index
+
+To solve this issue, my idea was to change the `sitemap_generator` config to instead of generate just one `sitemap.xml` file, it would create two sitemaps (one for the blog and one for the website) and add them to a index file.
+
+
+This is how the `config/sitemap.rb` file needs to be changed:
+
+```
+blog_sitemap_opts = {
+  create_index: false,
+  default_host: 'https://fastruby.io/blog',
+  compress: false,
+  sitemaps_path: '',
+  namer: SitemapGenerator::SimpleNamer.new(:blog_sitemap)
+}
+
+SitemapGenerator::Sitemap.create blog_sitemap_opts do
+  pages = Dir["public/blog/**/*.html"]
+  pages.each do |blog_page|
+    add blog_page, changefreq: 'weekly'
+  end
+end
+
+sitemap_opts = {
+  create_index: false,
+  default_host: 'https://fastruby.io',
+  compress: false,
+  sitemaps_path: '',
+  namer: SitemapGenerator::SimpleNamer.new(:sitemap)
+}
+
+SitemapGenerator::Sitemap.create sitemap_opts do
+  add '/#contact-us', changefreq: 'weekly'
+  add '/team', changefreq: 'weekly'
+
+  # all other important page here
+
+  add_to_index "blog_sitemap.xml", host: ENV['SITE_URL']
+end
+
+```
+
+As you can notice, in the first `SitemapGenerator::Sitemap.create` block we are listing all files inside the `blog` folder and adding their path to the blog_sitemap file.
+
+After that we are generating the `sitemap` for the website itself, listing all important pages and adding the `blog_sitemap` to the index.
+
+You can see the sitemap index file resulted by this config [at this link](http://fastruby.io.s3.amazonaws.com/sitemap.xml). It would be like this:
+
+```
+<sitemapindex xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/siteindex.xsd">
+  <sitemap>
+    <loc>http://fastruby.io.s3.amazonaws.com/blog_sitemap.xml</loc>
+    <lastmod>2019-12-02T17:54:36+00:00</lastmod>
+  </sitemap>
+  <sitemap>
+    <loc>http://fastruby.io.s3.amazonaws.com/sitemap1.xml</loc>
+    <lastmod>2019-12-02T17:54:36+00:00</lastmod>
+  </sitemap>
+</sitemapindex>
+```
+
+## And this is it!
+
+We know that for a better SEO, it is important to have a `sitemap` file. But since this is a static file sometimes it's a little tricky to keep them updated. So I just described my solution for that issue.
```

Comment:
```suggestion
We know that, for better SEO, it is important to have a `sitemap` file. But since this is a static file sometimes it's a little tricky to keep it updated. So I just described my solution to that issue.
```